django_project/changes/tests/test_github_pull_request.py

```diff
@@ -236,10 +249,32 @@ def test_download_all_referenced_images(self):
             content_type='application/json',
             HTTP_X_REQUESTED_WITH='XMLHttpRequest'
         )
+
+        entry = self.Entry.objects.last()
+        self.assertEqual(response.status_code, 200)
         self.assertJSONEqual(
             str(response.content, encoding='utf8'),
             {'status': 'success'}
         )
-        self.assertFalse('/media/media' in self.entry.image_file.url)
-        self.assertTrue('/media/images/entries' in self.entry.image_file.url)
-        self.assertFalse('<img  src="" />' in self.entry.description)
+        self.assertTrue(entry.image_file)
+
+        self.assertIn(
+            '106831433-dea95b80-66ca-11eb-8026-6823084d726e',
+            entry.image_file.name
+        )
+
+        image_file_dirname = os.path.dirname(entry.image_file.name)
+        image_file_upload_to = re.sub(r'^/', '', image_file_dirname)
+        image_file_url = settings.MEDIA_URL + image_file_upload_to
+
+        # check if upload_to give the same path
+        self.assertEqual(
+            entry.image_file.field.upload_to,
+            image_file_upload_to
+        )
+
+        # check if upload_to give the same path
+        self.assertEqual(
+            os.path.dirname(entry.image_file.url),
+            image_file_url,
+        )
```

Comment:
Thank you @lucernae for reviewing. I will partially update this PR since I need to digest the review and learn the solution as per your suggestion :sweat_smile:  

> Also, I'm not sure why you want to compare only the dirname and not the image name too.

when I do test on my localmachine, the image name will be
```images/entries/106831433-dea95b80-66ca-11eb-8026-6823084d726e-148.png``` 
instead of 
```images/entries/106831433-dea95b80-66ca-11eb-8026-6823084d726e.png```
 
There will be suffix `-n` in filename prior to the extension file comes from 
https://github.com/kartoza/prj.app/blob/dbf642b9185448f582fd42d3c1f28a81336b1806/django_project/changes/views/changelog_github.py#L271

Reply:
so that I don't know what the value should be compare with filename since the `-n` will give int number that I don't have idea how to address it.